File: set1/task0.m
a = linspace(1, 100, 100)
b = linspace(1, 99, 50)
c = linspace(-1.0, 1.0, 201).*pi
d = cat(2, linspace(-1.0, -0.01, 100), linspace(0.01, 1.0, 100)).*pi

e_ = sin(linspace(1, 100, 100));
e = e_ .* (e_ > 0);

A = reshape(a, [10, 10])
dia = linspace(99, 1, 99);
B = diag((1:100)) + diag(dia, -1) + diag(dia, 1)
C = triu(ones(10, 10))

D = cat(1, arrayfun(@(x) (x * (x + 1)) / 2.0, a), factorial(a))

n = 10;
E = ones(n, n);

for i = 1:n
    for j = 1:n
        if mod(j, i) ~= 0
            E(i, j) = 0;
        end
    end
end

E

File: set2/task0.py
import numpy as np

a = np.arange(1, 101, 1)
b = np.arange(1, 100, 2)
_c = np.arange(-1.0, 1.01, 0.01)
c = _c * np.pi
d = _c[(_c > 0.0001) | (_c < -0.0001)] * np.pi
e = np.maximum(np.sin(a), np.zeros(a.shape))

for v in (a, b, c, d, e):
    print(v)
print(c.shape, d.shape)

A = a.reshape((10, 10))
d1 = np.arange(99, 0, -1)
B = np.diag(d1, -1) + np.diag(a) + np.diag(d1, 1)
C = np.triu(np.ones((100, 100)))
D = np.array((np.arange(1, 11), np.arange(1, 11)))
for i in range(1, D.shape[1]):
    D[:, i] = [i + 1 + D[0, i - 1], (i + 1) * D[1, i - 1]]
    
E = np.array([[1 if i % j == 0 else 0 for j in range(1, 101)] for i in range(1, 101)])

for m in (A, B, C, D, E):
    print(m)